### ISSUE TITLE
GraphGradCAMExplainer use of backpropogation

### ISSUE THREAD

**Issue by CarlinLiao** [BiomedSciAI/histocartography#36]
When using the GraphGradCAMExplainer, we use an pretrained torch GNN model set to eval mode since we're no longer training the model. However, to find the node importances, the Explainer module uses backpropogation to find the node importances via the weight coefficients of the hooked activation maps, which shouldn't be possible on an eval model instance.

<img width="1059" alt="image" src="https://user-images.githubusercontent.com/12687873/191588014-d957f77f-59fc-4693-bade-8cca8a481c19.png">

For whatever reason, this doesn't throw an error in the recommended python 3.7, dgl 0.4.3post2, and torch 1.10 environment, but does in my more up-to-date python 3.9, dgl 0.9, torch 1.12.1 env even though the written code is identical.

The only solution I've found so far is to set the model used in the Explainer to training mode before running the explainer, but that's far from ideal.

Is there a way to find the node importances without committing to backpropogation? Is that what backpropogating in the original histocartography environment does instead? If it doesn't, is it not an issue that the model is being updated via backpropogation during the process of explaining node importance?
